engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -130,11 +130,13 @@
 	 */
 	private final ValidatorScopedContext validatorScopedContext;
 
+	private final ScriptEvaluatorFactory scriptEvaluatorFactory;
+
 	/**
 	 * The constraint initialization context is stored at this level to prevent creating a new instance each time we
 	 * initialize a new constraint validator as, for now, it only contains Validator scoped objects.
 	 */
-	private final HibernateConstraintValidatorInitializationContext constraintValidatorInitializationContext;
+	private HibernateConstraintValidatorInitializationContext constraintValidatorInitializationContext;
```

Comment:
A wise man said to me one day: "Every time you remove a `final`, God kills a kitten." (yes, he's so wise he uses backticks when he speaks).

You're paving the way to concurrency issues. Keep in mind that the `ValidatorImpl` is shared across thread.